diff --git a/mlir/include/mlir/IR/Threading.h b/mlir/include/mlir/IR/Threading.h
index 4dc067667..753503d3e 100644
--- a/mlir/include/mlir/IR/Threading.h
+++ b/mlir/include/mlir/IR/Threading.h
@@ -39,43 +39,10 @@ LogicalResult failableParallelForEach(MLIRContext *context, IteratorT begin,
   if (numElements == 0)
     return success();
 
-  // If multithreading is disabled or there is a small number of elements,
-  // process the elements directly on this thread.
-  if (!context->isMultithreadingEnabled() || numElements <= 1) {
-    for (; begin != end; ++begin)
-      if (failed(func(*begin)))
-        return failure();
-    return success();
-  }
-
-  // Build a wrapper processing function that properly initializes a parallel
-  // diagnostic handler.
-  ParallelDiagnosticHandler handler(context);
-  std::atomic<unsigned> curIndex(0);
-  std::atomic<bool> processingFailed(false);
-  auto processFn = [&] {
-    while (!processingFailed) {
-      unsigned index = curIndex++;
-      if (index >= numElements)
-        break;
-      handler.setOrderIDForThread(index);
-      if (failed(func(*std::next(begin, index))))
-        processingFailed = true;
-      handler.eraseOrderIDForThread();
-    }
-  };
-
-  // Otherwise, process the elements in parallel.
-  llvm::ThreadPool &threadPool = context->getThreadPool();
-  llvm::ThreadPoolTaskGroup tasksGroup(threadPool);
-  size_t numActions = std::min(numElements, threadPool.getThreadCount());
-  for (unsigned i = 0; i < numActions; ++i)
-    tasksGroup.async(processFn);
-  // If the current thread is a worker thread from the pool, then waiting for
-  // the task group allows the current thread to also participate in processing
-  // tasks from the group, which avoid any deadlock/starvation.
-  tasksGroup.wait();
-  return failure(processingFailed);
+  for (; begin != end; ++begin)
+    if (failed(func(*begin)))
+      return failure();
+  return success();
 }
 
 /// Invoke the given function on the elements in the provided range
