diff --git a/mlir/examples/toy/Ch7/toyc.cpp b/mlir/examples/toy/Ch7/toyc.cpp
index f10a95b724ca..34c7c617b0c2 100644
--- a/mlir/examples/toy/Ch7/toyc.cpp
+++ b/mlir/examples/toy/Ch7/toyc.cpp
@@ -44,7 +44,7 @@ namespace cl = llvm::cl;
 
 static cl::opt<std::string> inputFilename(cl::Positional,
                                           cl::desc("<input toy file>"),
-                                          cl::init("-"),
+                                          cl::init("input.toy"),
                                           cl::value_desc("filename"));
 
 namespace {
@@ -78,7 +78,8 @@ static cl::opt<enum Action> emitAction(
     cl::values(clEnumValN(DumpLLVMIR, "llvm", "output the LLVM IR dump")),
     cl::values(
         clEnumValN(RunJIT, "jit",
-                   "JIT the code and run it by invoking the main function")));
+                   "JIT the code and run it by invoking the main function")),
+    cl::init(Action::DumpMLIRLLVM));
 
 static cl::opt<bool> enableOpt("opt", cl::desc("Enable optimizations"));
 
@@ -275,7 +276,9 @@ int main(int argc, char **argv) {
   // If we aren't exporting to non-mlir, then we are done.
   bool isOutputingMLIR = emitAction <= Action::DumpMLIRLLVM;
   if (isOutputingMLIR) {
-    module->dump();
+    std::error_code EC;
+    llvm::raw_fd_ostream output_stream("output.mlir", EC);
+    module->print(output_stream);
     return 0;
   }
 
